### PR TITLE
Add `DbQueryCore` phpunit test 

### PR DESCRIPTION
**Pull request by PululuK**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop 
| Description?      | Add `DbQueryCore` some phpunit test.
| Type?             | refacto
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | N/A
| How to test?      | Tests green ;)
| Possible impacts? | N/A

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/24592)

### CODE REVIEW

File: classes/db/DbQuery.php [hunk 1]
```diff
@@ -331,4 +331,14 @@ public function __toString()
     {
         return $this->build();
     }
+
+    /**
+     * Get query.
+     *
+     * @return array
+     */
+    public function getQuery()
```
Comment (Progi1984) [inline classes/db/DbQuery.php]:
```suggestion
    public function getQuery(): array
```

File: tests/Unit/Classes/Db/DbQueryCoreTest.php [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+namespace Tests\Unit\Classes;
+
+if (!defined('_DB_PREFIX_')) {
+    define('_DB_PREFIX_', 'test_');
+}
+
+use DbQuery;
+use PHPUnit\Framework\TestCase;
+
+class DbQueryCoreTest extends TestCase
```
Comment (Progi1984) [inline tests/Unit/Classes/Db/DbQueryCoreTest.php]:
```suggestion
class DbQueryTest extends TestCase
```

File: tests/Unit/Classes/Db/DbQueryCoreTest.php [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+{
+    /*
+     * get DbQuery object
+     *
+     * @return DbQuery
+     */
+    public function dbQueryInstance(): DbQuery
```
Comment (Progi1984) [inline tests/Unit/Classes/Db/DbQueryCoreTest.php]:
```suggestion
    private function getDbQueryInstance(): DbQuery
```

File: tests/Unit/Classes/Db/DbQueryCoreTest.php [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+    public function testType($type, string $expectedType): void
+    {
+        $dbQuery = $this->dbQueryInstance();
+        $dbQuery->type($type);
+        $this->assertSame($expectedType, $dbQuery->getQuery()['type']);
+    }
+
+    /**
+     * @param mixed $fields
+     * @param mixed $expectedSelect
```
Comment (Progi1984) [inline tests/Unit/Classes/Db/DbQueryCoreTest.php]:
```suggestion
     * @param array $expectedSelect
```

File: tests/Unit/Classes/Db/DbQueryCoreTest.php [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+     *
+     * @dataProvider providerSelect
+     */
+    public function testSelect($fields, $expectedSelect): void
```
Comment (Progi1984) [inline tests/Unit/Classes/Db/DbQueryCoreTest.php]:
```suggestion
    public function testSelect($fields, array $expectedSelect): void
```

File: tests/Unit/Classes/Db/DbQueryCoreTest.php [hunk 6]
```diff
@@ -0,0 +1,186 @@
...
+    }
+
+    /**
+     * @param string $table
+     * @param string|null $alias
+     * @param string $expectedValue
+     *
+     * @dataProvider providerFrom
+     */
+    public function testFrom($table, $alias, $expectedValue): void
```
Comment (Progi1984) [inline tests/Unit/Classes/Db/DbQueryCoreTest.php]:
```suggestion
    public function testFrom(string $table, ?string $alias, string $expectedValue): void
```

File: tests/Unit/Classes/Db/DbQueryCoreTest.php [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+            ['table_name', 'alias', [
+                0 => '`' . _DB_PREFIX_ . 'table_name` alias',
+            ]],
+            ['table_name', null, [
+                0 => '`' . _DB_PREFIX_ . 'table_name`',
+            ]],
+        ];
+    }
+
+    public function providerBuild()
```
Comment (Progi1984) [inline tests/Unit/Classes/Db/DbQueryCoreTest.php]:
```suggestion
    public function providerBuild(): array
```

File: tests/Unit/Classes/Db/DbQueryCoreTest.php [hunk 8]
```diff
@@ -0,0 +1,186 @@
...
+namespace Tests\Unit\Classes;
+
+if (!defined('_DB_PREFIX_')) {
+    define('_DB_PREFIX_', 'test_');
+}
+
+use DbQuery;
+use PHPUnit\Framework\TestCase;
+
+class DbQueryTest extends TestCase
```
Comment (Progi1984) [inline tests/Unit/Classes/Db/DbQueryCoreTest.php]: @PululuK Could you rename the file too?

File: tests/Unit/Classes/Db/DbQueryCoreTest.php [hunk 9]
```diff
@@ -0,0 +1,186 @@
...
+            ]],
+            ['table_name', null, [
+                0 => '`' . _DB_PREFIX_ . 'table_name`',
+            ]],
+        ];
+    }
+
+    public function providerBuild(): array
+    {
+        define('_BREAK_LINE_', "\n");
```
Comment (PierreRambaud) [inline tests/Unit/Classes/Db/DbQueryCoreTest.php]: Use class const instead, defining a constant is useless :)

Comment (PierreRambaud) [reply tests/Unit/Classes/Db/DbQueryCoreTest.php]: This is also something you can refactor in the DbQueryCore class by using PHP_EOL or another const

File: tests/Unit/Classes/Db/DbQueryTest.php [hunk 10]
```diff
@@ -0,0 +1,186 @@
...
+if (!defined('_DB_PREFIX_')) {
+    define('_DB_PREFIX_', 'test_');
+}
+
+use DbQuery;
+use PHPUnit\Framework\TestCase;
+
+class DbQueryTest extends TestCase
+{
+    public const _BREAK_LINE_ = "\n";
```
Comment (PierreRambaud) [inline tests/Unit/Classes/Db/DbQueryTest.php]:
```suggestion
    public const BREAK_LINE = "\n";
```

**Review by PierreRambaud — APPROVED** [review]
Great job :+1:

**Review by Progi1984 — APPROVED** [review]

**Comment by Progi1984** [comment]
No need QA : it's only tests

**Pull request merged**